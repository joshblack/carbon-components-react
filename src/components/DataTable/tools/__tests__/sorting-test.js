import { getCellId } from '../';
import {
  sortStates,
  initialSortState,
  getSortState,
  defaultSortRows,
} from '../sorting';

describe('sorting tools', () => {
  it('should export an object of sorting states', () => {
    expect(sortStates).toMatchSnapshot();
  });

  it('should define an initial sort state', () => {
    expect(initialSortState).toBeDefined();
    expect(initialSortState).toBe(sortStates.NONE);
  });

  describe('#getSortState', () => {
    it('should cycle through sortStates when on the same header', () => {
      expect(getSortState('a', 'a', sortStates.NONE)).toBe(sortStates.DESC);
      expect(getSortState('a', 'a', sortStates.DESC)).toBe(sortStates.ASC);
      expect(getSortState('a', 'a', sortStates.ASC)).toBe(sortStates.NONE);
      expect(getSortState('a', 'a', sortStates.NONE)).toBe(sortStates.DESC);
    });

    it('should reset the state when switching headers', () => {
      expect(getSortState('a', 'a', sortStates.NONE)).toBe(sortStates.DESC);
      expect(getSortState('a', 'b', sortStates.DESC)).toBe(sortStates.DESC);
    });
  });

  describe('#defaultSortRows', () => {
    let mockArgs;
    let mockHeader;

    beforeEach(() => {
      mockHeader = 'header';
      mockArgs = {
        rowIds: ['a', 'b', 'c'],
        cellsById: {
          [getCellId('a', mockHeader)]: {
            value: 'c',
          },
          [getCellId('b', mockHeader)]: {
            value: 'b',
          },
          [getCellId('c', mockHeader)]: {
            value: 'a',
          },
        },
        direction: sortStates.ASC,
        key: mockHeader,
        locale: 'en',
      };
    });

    it('should sort the given rows', () => {
      expect(defaultSortRows(mockArgs)).toMatchSnapshot();
      expect(
        defaultSortRows({ ...mockArgs, direction: sortStates.ASC })
      ).toMatchSnapshot();
    });
  });
});
