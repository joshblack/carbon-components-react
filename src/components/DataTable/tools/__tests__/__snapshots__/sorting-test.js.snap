// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sorting tools #defaultSortRows should sort the given rows 1`] = `
Array [
  "c",
  "b",
  "a",
]
`;

exports[`sorting tools #defaultSortRows should sort the given rows 2`] = `
Array [
  "c",
  "b",
  "a",
]
`;

exports[`sorting tools should export an object of sorting states 1`] = `
Object {
  "ASC": "ASC",
  "DESC": "DESC",
  "NONE": "NONE",
}
`;
